src/Microsoft.Framework.Runtime.Hosting/Json/JsonBuffer.cs

```diff
@@ -0,0 +1,312 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Framework.Runtime.Json
+{
+    internal class JsonBuffer
+    {
+        private readonly TextReader _reader;
+        private JsonToken _token;
+        private int _line;
+        private int _column;
+
+        public JsonBuffer(TextReader reader)
+        {
+            _reader = reader;
+            _line = 1;
+        }
+
+        public JsonToken Read()
+        {
+            int first;
+            while (true)
+            {
+                first = ReadNextChar();
+
+                if (first == -1)
+                {
+                    _token.Type = JsonTokenType.EOL;
+                    return _token;
+                }
+                else if (!IsWhitespace(first))
+                {
+                    break;
+                }
+            }
+
+            _token.Value = null;
+            _token.Line = _line;
+            _token.Column = _column;
+
+            if (first == '{')
+            {
+                _token.Type = JsonTokenType.LeftCurlyBracket;
+            }
+            else if (first == '}')
+            {
+                _token.Type = JsonTokenType.RightCurlyBracket;
+            }
+            else if (first == '[')
+            {
+                _token.Type = JsonTokenType.LeftSquareBracket;
+            }
+            else if (first == ']')
+            {
+                _token.Type = JsonTokenType.RightSquareBracket;
+            }
+            else if (first == ':')
+            {
+                _token.Type = JsonTokenType.Colon;
+            }
+            else if (first == ',')
+            {
+                _token.Type = JsonTokenType.Comma;
+            }
+            else if (first == '"')
+            {
+                _token.Type = JsonTokenType.String;
+                _token.Value = ReadString();
+            }
+            else if (first == 't')
+            {
+                ReadLiteral(JsonConstants.ValueTrue);
+                _token.Type = JsonTokenType.True;
+            }
+            else if (first == 'f')
+            {
+                ReadLiteral(JsonConstants.ValueFalse);
+                _token.Type = JsonTokenType.False;
+            }
+            else if (first == 'n')
+            {
+                ReadLiteral(JsonConstants.ValueNull);
+                _token.Type = JsonTokenType.Null;
+            }
+            else if ((first >= '0' && first <= '9') || first == '-')
```

Comment:
It's complicate here. `0`, `0.1` are legal but `01` is illegal. The responsibility of parsing number is delegated to `JsonNumber`. Should we want to throw on the case of leading zero, we can throw there.

But I do wondering if it worth to detect this error.